2364/2/parser.rb

```diff
@@ -0,0 +1,48 @@
+require 'russian_obscenity'
+require 'terminal-table'
+require_relative 'participant.rb'
+require_relative 'create_data.rb'
+require_relative 'name_check.rb'
+require_relative 'obscene_rapers.rb'
+require_relative 'top_words.rb'
+
+# Class parse for parsing data
+class Parser
+  PATH = File.expand_path('.').freeze
+  attr_reader :participants, :methods
+  def initialize
+    @participants = []
+    @methods = create_methods
+    files_path = Dir[PATH + '/rap-battles/*']
+    create_participants(files_path)
+  end
+
+  def call(name, value)
+    if name
+      TopWords.new(value, name, participants).print_top_words
+    else
+      ObsceneRapers.new(value, participants).print_obscene_participants
+    end
+  end
+
+  private
+
+  def create_methods
```

Comment:
this is black magic, get rid of that proc logic or split it to separate classes and cover with tests